README.md

```diff
@@ -11,10 +11,6 @@
 https://user-images.githubusercontent.com/6180201/194214280-3a4106d5-53cf-4463-ba0e-201d8e1b0c0e.mp4
 
 
-# Contribution ideas
```

Comment:
Actually, can you upload a gif/video showing Whisper Playground in action?

Reply:
Following the latest commit, here is a demo that perfectly demonstrates the weaknesses of this approach to real-time diarization:

[Whisper Playground Demo](https://drive.google.com/file/d/1X4CMHY8u2pba7-487rr5wx-WEgfW_bXm/view?usp=sharing)

[Notebook for this run](https://colab.research.google.com/drive/1HIdkG2RlIi58C9Q36CZva9z10DmJ6UNu?usp=sharing)

1. False speaker re-identification: This is especially common at the beginning when there hasn't been a chance for the Diart to improve the embedding of an existing speaker due to the lack of speech. However, I believe this can be improved by implementing a speech-embedding brain that can extract more generalized embeddings with a small amount of data (e.g. SpeechBrain). However, it's been shown that this would lead to substantially slower execution time, so it's a choice between accuracy and efficiency (users would be able to choose the embedding brain). Gladly, Diart also supports custom embedding models out-of-the-box.

Refer to: [SpeechBrain Embedding Compatibility with Diart](https://github.com/juanmc2005/diart/issues/34)

2. Wrong speaker identification: Following 1, when there's an immediate new speaker identification, the audio that follows may be attributed to that new speaker as well. This can be seen in the second line of the demo video, for Speaker 2. For reference, there are only 2 speakers. Also happens in the third line.


Transcription weaknesses:

1. Difficulty with incorporating other implementations of Whisper: Diart anchors to Python 3.8, which already provides a challenge with incorporating libraries such as WhisperX, which anchors to 3.10. However, this is something I will explore further and gradually add support for the latest implementations.

2. Alignment failures: For some reason, I've had stable-ts fail, having to default to the original transcription. Sometimes, I had an empty transcription. Not knowing whether it came from the original transcription made with faster-whisper or from the aligned version, I default to the original transcription provided by faster-whisper.


Transcription strengths:

Device: CUDA
Compute type: int8_float16

1. Higher batch size with near-immediate response: As can be seen in the demo, I chose a beam size of 5, which is quite high compared to the default beam size of 1. This allows for much better accuracy, and thanks to faster-whisper, the transcription output remains near-immediate.

2. Lower memory usage: As can be seen in the [faster-whisper repo](https://github.com/guillaumekln/faster-whisper) provided benchmarks, faster-whisper cuts down the memory usage by an astonishing amount, especially when compared to the original Whisper implementation by OpenAI.

Reply:
Following this - performance for the sequential option.

[Demo](https://drive.google.com/drive/u/0/my-drive)

[Notebook](https://colab.research.google.com/drive/1HIdkG2RlIi58C9Q36CZva9z10DmJ6UNu?usp=sharing)

Bugs:

The model will occasionally swap between the speaker identifiers, currently, no way to fix this other than take the matter of managing speakers into our own hands, drastically increasing the complexity of the process. 

Pros:

- Accuracy and refinement: Thanks to the re-iterations, the punctuation changes as more transcriptions are generated to better fit the transcription, and incorrect words tend to quickly be corrected as more data is available, we could call this "constant refinement".


Cons:

- Speed: Even though we're using faster-whisper, the decrease in speed the more a person talks is very much noticeable compared to the real-time option. This is to be expected, as it would transcribe two minutes of audio for a stream of two minutes. And then it would transcribe audio of two minutes + transcription timeout time, and so on.

- Segment-level rather than word-level: This is the starting point, which is also why the diarization works at segment-level rather than word-level. It manages fine though, considering it has a lot of data to work with and infinite re-iterations. Simply using the readily-available pipeline, no use of Pyannote's building blocks. If we did use those, we could also better diarizate the occasional comments better.


Bugs in the latest commit:

- Can't run it locally, can only run it in a Google Colab environment. Getting quite a weird error that seems to appear when Diart is installed. A clash between av files. MKL is also disabled upon setup when using the installation script. Hoping to get this sorted out.

Reply:
> Bugs in the latest commit:
> 
> Can't run it locally, can only run it in a Google Colab environment. Getting quite a weird error that seems to appear when Diart is installed. A clash between av files. MKL is also disabled upon setup when using the installation script. Hoping to get this sorted out.

@ethanzrd please open an issue and label it as `Bug`